### ISSUE TITLE
[Yaml] Quoting of dumped strings that would be considered boolean by YAML <1.2

### ISSUE THREAD

**Issue by petert82** [symfony/symfony#13209]
I understand the Yaml component partially implements YAML 1.2, and as such does not quote PHP strings containing values that would be considered as boolean values by parsers implementing older versions of the YAML spec.

For example, dumping this PHP array:

```php
array('country_code' => 'no');
``` 

Would output yaml like:

```yaml
country_code: no
```

And **not**:

```yaml
country_code: 'no'
```

This makes using yaml files dumped by the Yaml component in environments where they will be parsed by non-1.2 compliant parsers *interesting*.

Would it be desirable to change the behaviour of the `Dumper` so that it quotes [possibly-boolean-per-the-older-spec][1] strings (`'yes'`, `'no'`, `'on'`, `'off'`, etc)? Obviously this extra quoting is not required for YAML 1.2 compliance, but I don't *think* this change would have any harmful side-effects.

It looks like implementing this would be a simple change in the `Yaml\Inline` class - and I'd be happy to do it (1st time contributer, please be gentle :D) - but I don't really want to try patching this in if it's decided this is a stupid/impossible idea.

[1]: http://yaml.org/type/bool.html

**Comment by WouterJ** [comment]
:+1: 

Btw, I think you can better add these rules to https://github.com/symfony/Yaml/blob/master/Escaper.php#L73-76

**Comment by stof** [comment]
I think it is a good idea to escape them when dumping to increase the interoperability. Please submit a PR with the change

**Comment by petert82** [comment]
Ok, which branch should this change be on? Is this considered a bug fix and so 2.3?

**Comment by fabpot** [comment]
@petert82 I would consider it as a bug fix indeed, so 2.3.
